torch/csrc/jit/passes/quantization/insert_quant_dequant.cpp

```diff
@@ -224,6 +241,49 @@ bool isNoopObserver(Value* observer) {
   return false;
 }
 
+bool isFP16NoopObserver(script::Module& module, Node* n) {
+  Value* v = n->output();
+  auto observer = n->input(0);
+  auto observer_module = module.attr(findObserverName(v).value()).toModule();
+  return (observer_module.attr("dtype") == at::ScalarType::Half) &&
+      isNoopObserver(observer);
+}
+
+std::string getEmbeddingBagObsName(script::Module& module, Node* n) {
+  Value* v = n->output();
+  auto observer = n->input(0);
+  auto observer_module = module.attr(findObserverName(v).value()).toModule();
+  if (observer_module.hasattr("custom_op")) {
+    auto op_name = observer_module.attr("custom_op").toStringRef();
+    return isNoopObserver(observer) ? op_name : "";
```

Comment:
follow up PR: since NoopObserver is special, probably better to add a "_" prefix to reserve this for internal use.